examples/elasticsearch-incremental/elasticsearch-incremental-blueprint.yaml

```diff
@@ -0,0 +1,542 @@
+metadata: 
+  name: elasticsearch-incremental-blueprint 
+actions:
+  backup:
+    outputArtifacts:
+      s3Snap:
+        keyValue:
+          repoPath: '{{ .Phases.setupPhase.Output.repoPath }}'
+          snapshotName: '{{ .Phases.setupPhase.Output.snapshotName }}'
+          esClusterName: '{{ .Phases.setupPhase.Output.esClusterName }}'
+          esClusterNamespace: '{{ .Object.metadata.namespace }}'
+    phases:   
+    - func: KubeTask
+      name: setupPhase
+      objects:
+        elasticSecret:
+          kind: Secret
+          name: '{{ .Object.metadata.name }}-es-elastic-user'
+          namespace: '{{ .Object.metadata.namespace }}' 
+      args:        
+        image: ghcr.io/kanisterio/kanister-kubectl-1.18:0.81.0
+        command:
+        - bash
+        - -x
+        - -o
+        - errexit
+        - -o
+        - pipefail
+        - -c
+        - |
+          debug=0
+          function dbg
+          {
+              if [[ $debug -eq 1 ]]
+              then 
+                echo "$(date "+%Y-%m-%d %H:%M:%S.%3N") - $0 - $*" >> /tmp/k10_debug.log
+              fi
+          }
+          function bp_echo
+          {
+            # use 
+            # kubectl logs -n kasten-io -l component=kanister --tail=10000 -f | ggrep -o -P '(?<=Elasticbps3echobegin).*(?=Elasticbps3echoend)' 
+            # to grab the logs in kanisters 
+            echo "Elasticbps3echobegin backup-action: $* Elasticbps3echoend"
+          }            
+          PROFILE_TYPE="{{ .Profile.Location.Type }}"
+          if [[ $PROFILE_TYPE != "s3Compliant" ]]
+          then 
+            dbg "Only s3Compliant profile are supported, exiting"
+            exit 1
+          fi
+          CLUSTER_UID=$(kubectl get ns default -o jsonpath='{.metadata.uid}')
+          NS_UID=$(kubectl get ns {{ .Object.metadata.namespace }} -o jsonpath='{.metadata.uid}')
+          ES_CLUSTER_UID={{ .Object.metadata.uid }}
+          REPO_PATH=k10/$CLUSTER_UID/elastic-search/$NS_UID/$ES_CLUSTER_UID
+          SNAPSHOT_NAME="snap_{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time | date "2006-01-02t15:04:05z07:00" }}"
+          bp_echo "output the snapshot info $SNAPSHOT_NAME" 
+          kando output repoPath $REPO_PATH 
+          kando output snapshotName $SNAPSHOT_NAME 
+          kando output esClusterName {{ .Object.metadata.name }}
+
+          # setup the credentials on the keystore setting of each nodes 
+          {{- if .Profile.Credential.KeyPair }}
+          AWS_SECRET_KEY="{{ .Profile.Credential.KeyPair.Secret }}"
+          AWS_ACCESS_KEY="{{ .Profile.Credential.KeyPair.ID }}"
+          {{- else }}
+          AWS_SECRET_KEY="{{ .Profile.Credential.Secret.Data.aws_secret_access_key | toString }}"
+          AWS_ACCESS_KEY="{{ .Profile.Credential.Secret.Data.aws_access_key_id | toString }}"
+          {{- end }}
+          # No Support for more than one node set for the moment
+          NODE_SET_COUNT={{ (index .Object.spec.nodeSets 0).count }}
```

Comment:
should we fail if there are more than 1 node set?

Reply:
Good point I will add that.